stdlib/http/src/main/java/org/ballerinalang/net/http/ClientListener.java

```diff
@@ -0,0 +1,133 @@
+import org.wso2.transport.http.netty.contract.websocket.WebSocketCloseMessage;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketConnection;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketControlMessage;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketHandshaker;
+import org.wso2.transport.http.netty.contract.websocket.WebSocketTextMessage;
+
+import java.io.IOException;
+
+import static org.ballerinalang.net.http.WebSocketConstants.STATUS_CODE_ABNORMAL_CLOSURE;
+import static org.ballerinalang.net.http.WebSocketUtil.hasRetryConfig;
+import static org.ballerinalang.net.http.WebSocketUtil.reconnect;
+import static org.ballerinalang.net.http.WebSocketUtil.setCloseMessage;
+
+/**
+ * Client connector listener for WebSocket.
+ *
+ * @since 1.0.0
+ */
+public class ClientListener extends WebSocketClientConnectorListener {
+    private WebSocketOpenConnectionInfo connectionInfo;
+    private static final Logger logger = LoggerFactory.getLogger(ClientListener.class);
+
+    public void setConnectionInfo(WebSocketOpenConnectionInfo connectionInfo) {
+        this.connectionInfo = connectionInfo;
+    }
+
+    @Override
+    public void onHandshake(WebSocketHandshaker webSocketHandshaker) {
+        throw new WebSocketException("onHandshake and onOpen is not supported for WebSocket client service");
+    }
+
+    @Override
+    public void onMessage(WebSocketTextMessage webSocketTextMessage) {
+        try {
+            WebSocketDispatcher.dispatchTextMessage(connectionInfo, webSocketTextMessage);
+        } catch (IllegalAccessException e) {
+            // Ignore as it is not possible have an Illegal access
+        }
+    }
+
+    @Override
+    public void onMessage(WebSocketBinaryMessage webSocketBinaryMessage) {
+        try {
+            WebSocketDispatcher.dispatchBinaryMessage(connectionInfo, webSocketBinaryMessage);
+        } catch (IllegalAccessException e) {
+            // Ignore as it is not possible have an Illegal access
+        }
+    }
+
+    @Override
+    public void onMessage(WebSocketControlMessage webSocketControlMessage) {
+        try {
+            WebSocketDispatcher.dispatchControlMessage(connectionInfo, webSocketControlMessage);
+        } catch (IllegalAccessException e) {
+            // Ignore as it is not possible have an Illegal access
+        }
+    }
+
+    @Override
+    public void onMessage(WebSocketCloseMessage webSocketCloseMessage) {
+        ObjectValue webSocketClient = connectionInfo.getWebSocketEndpoint();
+        int statusCode = webSocketCloseMessage.getCloseCode();
+        if (hasRetryConfig(webSocketClient) && statusCode == STATUS_CODE_ABNORMAL_CLOSURE &&
+                reconnect(connectionInfo)) {
+            return;
+        } else {
+            if (hasRetryConfig(webSocketClient) && statusCode != STATUS_CODE_ABNORMAL_CLOSURE) {
+                logger.info("Couldn't connect to the server because the given server: " +
```

Comment:
Shall we rephrase the log? Something like
```suggestion
                logger.debug("Reconnect attempt not made because of close initiated by the server: " +
```